### PR TITLE
Play 2.6 upgrade

### PR DESCRIPTION
**Pull request by TBonnin**
## What does this change?
Migrate dotcom from Play 2.5 to Play 2.6

If you want to know more about Play 2.6 migration, see https://www.playframework.com/documentation/2.6.x/Migration26

This is a big PR. It is better to review it commit by commit I think.
The 2 biggest changes are 
- [controller changes](https://www.playframework.com/documentation/2.6.x/Migration26#scala-controller-changes) brought by 2.6: d0e128b
- Deprecation of `play.api.libs.concurrent.Execution.Implicits.defaultContext` which means an `ExecutionContext` needs to be injected everywhere it is required: 9738408

Akka HTTP is the default server engine in Play 2.6. However it is possible to get back to Netty, which I did in 89295b9. The scope of this PR is big enough without migrating to Akka http. We will move to it in a subsequent PR.

Ping: @alexduf (I want you opinion about 7cde5d1 for instance), @mchv and @guardian/dotcom-platform

## What is the value of this and can you measure success?
Up to date dependencies.

## Does this affect other platforms - Amp, Apps, etc?
It does affect amp

## Tested in CODE?
Yes

### CODE REVIEW

File: project/Prototypes.scala [hunk 1]
```diff
@@ -125,32 +125,38 @@ trait Prototypes {
...
+      .settings(frontendDependencyManagementSettings)
+      .settings(frontendCompilationSettings)
+      .settings(frontendTestSettings)
+      .settings(VersionInfo.settings)
+      .settings(libraryDependencies ++= Seq(macwire, commonsIo))
+      .settings(packageName in Universal := applicationName)
+      .settingSets(settingSetsOrder)
+      .settings(
+        mappings in Universal ++= (file("ui/dist") ** "*").get.map { f => f.getAbsoluteFile -> f.toString }
+      )
```
Comment (TBonnin) [inline project/Prototypes.scala]: This is only formatting change

**Comment by alexduf** [comment]
I had more a glance than a review, but this looks like amazing work @TBonnin, well done!

I also looked at 7cde5d18e038c0c3154a0b7e015e4ae0830365bb and couldn't see anything wrong as far as I'm concerned.

I particularly like how you handled the ExecutionContext migration by relying on the controller components.

Keeping netty for now also sounds like a sensible move so that going to prod seems less scary.

File: sport/app/rugby/conf/RugbyLifecycle.scala [hunk 2]
```diff
@@ -22,12 +22,12 @@ class RugbyLifecycle(
   override def start(): Unit = {
     jobs.deschedule("FixturesAndResults")
     jobs.schedule("FixturesAndResults", "5 0/30 * * * ?") {
-      rugbyStatsJob.fetchFixturesAndResults()
+      rugbyStatsJob.fetchFixturesAndResults
```
Comment (mchv) [inline sport/app/rugby/conf/RugbyLifecycle.scala]: I don't think they are related to play `2.6 but `()` is used in `scala` on methods that have a side effect and therefore should not be removed here I think, see https://docs.scala-lang.org/style/naming-conventions.html#parentheses

File: sport/app/rugby/conf/RugbyLifecycle.scala [hunk 3]
```diff
@@ -22,12 +22,12 @@ class RugbyLifecycle(
...
     }
 
     jobs.deschedule("GroupTables")
     jobs.schedule("GroupTables", "10 0/30 * * * ?") {
-      rugbyStatsJob.fetchGroupTables()
+      rugbyStatsJob.fetchGroupTables
```
Comment (mchv) [inline sport/app/rugby/conf/RugbyLifecycle.scala]: idem

File: sport/app/rugby/conf/RugbyLifecycle.scala [hunk 4]
```diff
@@ -49,8 +49,8 @@ class RugbyLifecycle(
     }
 
     akkaAsync.after1s {
-      rugbyStatsJob.fetchFixturesAndResults()
-      rugbyStatsJob.fetchGroupTables()
+      rugbyStatsJob.fetchFixturesAndResults
+      rugbyStatsJob.fetchGroupTables
```
Comment (mchv) [inline sport/app/rugby/conf/RugbyLifecycle.scala]: idem

File: static/src/stylesheets/module/football/_matches.scss [hunk 5]
```diff
@@ -141,9 +141,25 @@
...
+    .i, svg {
+        width: $show-more-icon-size;
+        height: $show-more-icon-size;
+        vertical-align: middle;
+        margin: auto;
+        position: absolute;
+        top: 0;
+        bottom: 0;
+        left: $show-more-icon-spacing;
+    }
```
Comment (mchv) [inline static/src/stylesheets/module/football/_matches.scss]: I am not sure those changes are related

Comment (TBonnin) [reply static/src/stylesheets/module/football/_matches.scss]: With javascript disabled, which is how integration tests run, the football `show more` button is being hidden, even though it shouldn't be.
With former version of Selenium (brought within Play 2.5), clicking an hidden element during tests was not a problem however new version of Selenium prevents any interactions with an element that is not visible.
This fix ensures:
- the button is visible
- the integration tests checking the behaviour of this button works

There is a cleaner way to fix this issue but it involves touching all `show more` button throughout the website. I will implement it in a subsequent PR

File: sport/app/rugby/conf/RugbyLifecycle.scala (hunk 2 above)
Comment (TBonnin) [reply sport/app/rugby/conf/RugbyLifecycle.scala]: Fair point. 
However it means method that only takes implicit parameters need to be declare this way:
```
def myMethod()(implicit something: Something)
```
which is quite ugly

File: common/test/recorder/HttpRecorder.scala [hunk 6]
```diff
@@ -6,14 +6,16 @@ import java.nio.ByteBuffer
...
 import common.ExecutionContexts
 import conf.Configuration
-import io.netty.handler.codec.http.HttpHeaders
-import org.asynchttpclient.{Response => AHCResponse}
-import org.asynchttpclient.uri.Uri
 import org.apache.commons.codec.digest.DigestUtils
 import play.api.libs.ws.WSResponse
-import play.api.libs.ws.ahc.AhcWSResponse
+import play.api.libs.ws.ahc.{AhcWSResponse, StandaloneAhcWSResponse}
+import play.shaded.ahc.org.asynchttpclient.{Response => ACHResponse}
```
Comment (jfsoul) [inline common/test/recorder/HttpRecorder.scala]: Minor: I think it should be `AHCResponse` as opposed to `ACHResponse`

File: common/app/googleAuth/OAuthLoginController.scala [hunk 7]
```diff
@@ -97,20 +97,20 @@ trait OAuthLoginController extends Controller with ExecutionContexts with implic
...
-class AuthCookie(cryptoConfig: CryptoConfig) extends Logging {
+class AuthCookie(httpConfiguration: HttpConfiguration) extends Logging {
 
   private val cookieName = "GU_PV_AUTH"
   private val oneDayInSeconds: Int = 86400
 
   def from(id: UserIdentity): Option[Cookie] = {
     val idWith30DayExpiry = id.copy(exp = (System.currentTimeMillis() / 1000) + oneDayInSeconds )
-    Some(Cookie(cookieName,  Crypto.encryptAES(Json.toJson(idWith30DayExpiry).toString, cryptoConfig.secret), Some(oneDayInSeconds)))
+    Some(Cookie(cookieName,  Crypto.encryptAES(Json.toJson(idWith30DayExpiry).toString, httpConfiguration.secret.secret), Some(oneDayInSeconds)))
```
[the remark this answers is not in the document]
Comment (TBonnin) [reply common/app/googleAuth/OAuthLoginController.scala]: ¯\_(ツ)_/¯

File: common/test/recorder/HttpRecorder.scala (hunk 6 above)
Comment (TBonnin) [reply common/test/recorder/HttpRecorder.scala]: 👍

**Review by jfsoul — APPROVED** [review]
This looks good to me, great work 🎉

**Comment by TBonnin** [comment]
I think I have ironed out all the issues. I am planning to deploy this first thing tomorrow morning.
Shoot if you have any concern?

**Comment by mchv** [comment]
@TBonnin great work 🥇

**Pull request merged**